### PR TITLE
Add CLI alias support

### PR DESCRIPTION
**Pull request by evanphx**
## Summary
- Add `[aliases]` section to `.miren/app.toml` for defining custom CLI command shortcuts
- Aliases expand before dispatch, support multi-word names (e.g. `"x console"`), and append extra user arguments
- `miren help alias` shows usage documentation with colored output and lists configured aliases
- Alias names are validated and must not shadow built-in commands

## Example

```toml
# .miren/app.toml
[aliases]
console = "app run bin/rails console"
tail = "logs app -f"
"x console" = "app run bin/rails console"
```

Then `miren console`, `miren tail -n 50`, and `miren x console` all work.

## Test plan
- [x] AppConfig parsing and validation tests for aliases (valid names, invalid names, empty targets)
- [x] Tokenizer tests (quoted strings, escapes, edge cases)
- [x] Alias expansion tests (no config, built-in command priority, no args)
- [x] Linter passes

### CODE REVIEW

File: cli/alias.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	for i := 0; i < len(s); i++ {
+		c := s[i]
+
+		if escaped {
+			current = append(current, c)
+			escaped = false
+			continue
+		}
+
+		if c == '\\' && inDouble {
```
Comment (phinze) [inline cli/alias.go:85]: `\` only escapes inside double quotes — bare `\` outside quotes is literal. Makes sense for these simple command strings, just flagging in case it comes up.

**Review by phinze — APPROVED** [review]
Cool stuff! Our hunch is aliases will eventually come from client config too (not just app.toml), and it's nice to have a way to see what's active, so we added `miren alias list`:

```
NAME            COMMAND                        SOURCE
console         app run bin/rails console      .miren/app.toml:4
rails console   app exec -i bin/rails console  .miren/app.toml:6
tail            logs app -f                    .miren/app.toml:5
```

Also noticed the "built-in command not expanded" test was returning early at the no-config check and never hitting the shadow path — went ahead and fixed that up too. Filed MIR-874 for a broader sweep on nice config parse errors (e.g. `help alias` silently swallows parse failures). A couple other small things below, all optional.

<3 P 🤓 & C 🤖

File: cli/alias.go (hunk 1 above)
Comment (evanphx) [reply cli/alias.go:85]: k!
